wagtail/wagtailcore/fields.py

```diff
@@ -21,7 +21,24 @@ def formfield(self, **kwargs):
         return super(RichTextField, self).formfield(**defaults)
 
 
-class StreamField(with_metaclass(models.SubfieldBase, models.Field)):
+# https://github.com/django/django/blob/64200c14e0072ba0ffef86da46b2ea82fd1e019a/django/db/models/fields/subclassing.py#L31-L44
+class Creator(object):
+    """
+    A placeholder class that provides a way to set the attribute on the model.
+    """
+    def __init__(self, field):
+        self.field = field
+
+    def __get__(self, obj, type=None):
+        if obj is None:
+            return self
+        return obj.__dict__[self.field.name]
+
+    def __set__(self, obj, value):
+        obj.__dict__[self.field.name] = self.field.to_python(value)
+
+
```

Comment:
@mvantellingen This class is reinstating behaviour that previously existed in models.SubfieldBase but is removed in Django 1.10 - so now we're consistently using this copy of the code on all versions including Django 1.10.

(As per the discussion on #2119, though, we ought to investigate whether we really need this behaviour at all, or whether it can be dropped.)